coalib/results/TextPosition.py

```diff
@@ -31,3 +46,35 @@ def line(self):
     @property
     def column(self):
         return self._column
+
+    @classmethod
+    def from_position(cls, text, position, create_object=True):
+        """
+        Makes a TextPosition object or a tuple containing (line, column)
+        by calculating line number and column in the text, from position.
+        Uses \\n as the newline character. Lines and columns start from 1.
+
+        :param text:        A string with all the contents of file.
+        :param pos_to_find: Position of character to be found in the
+                            (line, column) form.
+        :create_object:     Boolean which determines whether to return
+                            a TextPosition Object or a tuple.
+        :return:            a TextPosition object or a tuple of the form
+                            (line, column).
+        """
+        line = 1
+        try:
+            while True:
+                textline = text[line - 1]
```

Comment:
It seems that only this line can produce an exception, so it'll be better to wrap only it with `try` clause. IMO it'll be easier to read, because you'll have to think less on what can produce an exception.